test/cli/help_test.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// @dart=2.10
+import 'package:args/command_runner.dart';
+import 'package:pub/src/command_runner.dart' show PubCommandRunner;
+import 'package:test/test.dart';
+
+import '../golden_file.dart';
+import '../test_pub.dart';
+
+/// Extract all commands and subcommands.
+///
+/// Result will be an iterable of lists, illustrated as follows:
+/// ```
+/// [
+///   [pub]
+///   [pub, get]
+///   ...
+/// ]
+/// ```
+Iterable<List<String>> _extractCommands(
+  List<String> parents,
+  Iterable<Command> cmds,
+) sync* {
+  if (parents.isNotEmpty) {
+    yield parents;
+  }
+  // Track that we don't add more than once, we don't want to test aliases
+  final names = <String>{};
+  yield* cmds
+      .where((sub) => !sub.hidden && names.add(sub.name))
```

Comment:
Maybe we should even do hidden commands - their help might be important to those who knows about it?